packages/components/pie-modal/test/visual/pie-modal.spec.ts

```diff
@@ -143,32 +183,119 @@ test.describe('`hasBackButton`', () => {
     });
 });
 
-test('long heading renders correctly', async ({ page, mount }) => {
-    await mount(PieModal, {
-        props: {
-            heading: 'This is a modal heading but super long and should span multiple lines, hopefully this should never happen on production!',
-            isOpen: true,
-            size: 'medium',
-            hasBackButton: true,
-            isDismissible: true,
-        },
+test.describe('Prop: `heading`', () => {
+    test('should display & render long headings correctly', async ({ page, mount }) => {
+        await mount(PieModal, {
+            props: {
+                heading: 'This is a modal heading but super long and should span multiple lines, hopefully this should never happen on production!',
+                isOpen: true,
+                size: 'medium',
+                hasBackButton: true,
+                isDismissible: true,
+                leadingAction: {
+                    text: 'Confirm',
+                    variant: 'primary',
+                    ariaLabel: 'Confirmation text',
+                },
+            } as ModalProps,
+        });
+
+        await percySnapshot(page, 'Modal - Long heading');
     });
+});
+
+test.describe('Prop: `isLoading`', () => {
+    test('should display loading spinner when `isLoading` is true', async ({ mount, page }) => {
+        await mount(PieModal, {
+            props: {
+                heading: 'This is a modal heading',
+                hasBackButton: true,
+                isDismissible: true,
+                isOpen: true,
+                isLoading: true,
+                leadingAction: {
+                    text: 'Confirm',
+                    variant: 'primary',
+                    ariaLabel: 'Confirmation text',
+                },
+            } as ModalProps,
+        });
 
-    await percySnapshot(page, 'Modal - Long heading');
+        await percySnapshot(page, `Modal displays loading spinner - isLoading: ${true}`);
+    });
 });
 
-test('Should display loading spinner when `isLoading` is true', async ({ mount, page }) => {
-    await mount(PieModal, {
-        props: {
-            heading: 'This is a modal heading',
-            hasBackButton: true,
-            isDismissible: true,
-            isOpen: true,
-            isLoading: true,
-        } as ModalProps,
+test.describe('Prop: `leadingAction`', () => {
+    test.describe('when prop is passed into component', () => {
+        test('should display `leadingAction` when props are passed correctly', async ({ mount, page }) => {
+            await mount(PieModal, {
+                props: {
+                    heading: 'This is a modal heading',
+                    hasBackButton: true,
```

Comment:
I don't think we need `hasBackButton` or `isDismissible` in any of these tests, those should be covered by our other tests